### ISSUE TITLE
use NekoX source code as the primary base for Telegraher

### ISSUE THREAD

**Issue by DI555** [nikitasius/Telegraher#13]
i would like to suggest to use NekoX source code as the primary base for Telegraher. It’s much better than using TelegramFOSS source code if you planned it! NekoX itself uses code from TelegramFOSS! But NekoX contains much more fetures than just TelegramFOSS!

Many thanks in advance.

**Comment by DI555** [comment]
imho, even some ideas might seem "bad", they’re optional there, and the power in it 👍👍👍
nekox is the huge opensourced tg combain that i’ve ever seen, so the unused powered options in one local area might be usefull in others areas instead! (for example the proxy support much useful in ch area and asia, etc.) and that bunch of variouse options will obviously bring to the Telegraher much more donations, if you plan to open donate it 👍
i mean, as for me, i don’t see really "bad" options there in nekox, and if any ones are unsuitable i just switch them off , because of optionability ;)

**Comment by DI555** [comment]
only one bad thing is - nekox team took decision to serve official tg policy, so any unofficial features in nekox will be unfortunately deprecated (((
so, Telegraher dev may concentrate only on any variouse unofficial features, like i saw - it planned this way from the start 👍

**Comment by Bugaddr** [comment]
Best will be to pick good things from nekox rather than rebasing to fork, also it makers things complicated for developer.

**Comment by nikitasius** [comment]
> only one bad thing is - nekox team took decision to serve official tg policy

actually they are attacking us. So i asked few questions (shared screens in the chat) and waiting the answers.

**Comment by nikitasius** [comment]
i mean the TG, not the neko :)

**Comment by DI555** [comment]
@nikitasius , yep they are angry (( , the wayout would be a custom tg servers imho..

ok, seams, neko is really complicated((
what do you think about TelegramFOSS as a primary base for Telegraher plus much of NekoX’s , Nekogram’s ,Forkgram’s , Nullgram’s , GuGugramX’s , exteraGram’s , OwlGram’s , telegram_foss_fork’s and Nagram’s features?

**Comment by DI555** [comment]
am i right, NekoX is a foss fork of Nekogram?
Neko started when Nekogram was public builds not created from Nekogram sources, is it true?
i mean, if NekoX sources are much complicated for Telegraher, may be sources of Nekogram from gitlab could be easier handling for Telegraher base?

**Comment by nikitasius** [comment]
> may be sources of Nekogram from gitlab could be easier handling for Telegraher base

Hi, i don't want to use feature pack of nekox or nekogram (1st from foss, 2nd from vanilla) as a base of telegraher.
These features are depends of these devs, if someone will be removed, it will be removed and more stuff in the client, harder to place our own.

Same time we use the vanilla keys. And the app do the vanilla behaviour. Using nekox or nekogram base app will not have the vanilla behaviour.

Next, these forks aren't official forks. If these devs will disappear (they are from China, it's non-free country). Noone will develop their works with the same speed and level, while vanilla TG still here (otherwise TG decide to stop developping vanilla client).

And, again, we use vanilla keys. Nekox use own keys, nekogram use own keys too. They could be blocked, while vanilla keys thats much harder (=impossible) to block.

**Comment by DI555** [comment]
and Telegram-FOSS uses non the vanilla keys, is it true ?

**Comment by nikitasius** [comment]
> and Telegram-FOSS uses non the vanilla keys, is it true ?

idk i did not reversed them. But if you think that picking some ideas can change our behavior - it's not totally correct. We still on vanilla core and we acting as an app w/o gapps on the phone but we have no google services inside the app, except the mltoolkit for translations.

**Comment by DI555** [comment]
yep, i see, the right way clearly in using vanilla core plus patches !!!
so, trecking and updating patches give much mobility and independency !!!

could we start from Telegram-FOSS bunch of patches ?
i didn’t see patches in much featured NekoX and Nekogram repos, but it’ll be imo no problem to convert their commits into patch forms, right ?

**Comment by DI555** [comment]
but, i like the NekoXer idea !!!
NekoX is so much featured, and so hard to miss evene one feature of it))
btw, the vanilla keys could be injected into the NekoXer, that way it may become a vanilla client ?

**Comment by nikitasius** [comment]
> but, i like the NekoXer idea !!! NekoX is so much featured, and so hard to miss even one feature of it)) btw, the vanilla keys could be problemlessly injected into the NekoXer? ..that way may it become like a vanilla client ?

i don't want to be dependant from 3rd party fork and his devs.
